### PR TITLE
[release-1.6] bump CAPI to v1.2.8

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind failing-test
/kind cleanup

**What this PR does / why we need it**: CAPI v1.2.8 added fixes for e2e tests related to the k8s.gcr.io -> registry.k8s.io migration and https://github.com/kubernetes-sigs/cluster-api/issues/7833. More context here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2978#issuecomment-1372636863.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Bump cluster-api dependency to v1.2.8
```

### CODE REVIEW

**Review by jackfrancis — APPROVED** [review]
/lgtm

File: go.mod [hunk 1]
```diff
@@ -211,4 +211,4 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.2.6
+replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.2.8
```
Comment (nojnhuh) [inline go.mod:214]: @CecileRobertMichon Do you know why this is here? It seems like a no-op and bit me when I was trying to update this.

Comment (CecileRobertMichon) [reply go.mod:214]: yes here's an explanation: https://github.com/kubernetes-sigs/cluster-api/issues/6081

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**